pkg/clustermanager/manager.go

```diff
@@ -468,6 +467,20 @@ func (m *Manager) UserContext(clusterName string) (*config.UserContext, error) {
 	return record.cluster, nil
 }
 
+// UserContextFromCluster accepts a pointer to a Cluster and returns a client
+// for that cluster. It does not start any controllers.
+func (m *Manager) UserContextFromCluster(cluster *v3.Cluster) (*config.UserContext, error) {
```

Comment:
Kudos for adding documentation comments.
I'm thinking if we could update func name to better convey the fact it doesn't start controllers. There is not much difference between `UserContext` and `UserContextFromCluster`, especially considering both take some info from cluster (either name or pointer to complete Cluster object). `UserContextNoControllers` or something like this?